### PR TITLE
Allow adding browserStack username and key in My Account

### PR DESCRIPTION
**Pull request by HelenaEksler**
#97

### CODE REVIEW

**Comment by HelenaEksler** [comment]
shoov5

**Comment by HelenaEksler** [comment]
selection_015
selection_016

File: client/app/scripts/controllers/account.js [hunk 1]
```diff
@@ -8,8 +8,25 @@
...
+    });
+
+    $scope.saveBrowserStackData = function() {
+      var data = {
+        'browserstack_username': $scope.account.browserstack_username,
+        'browserstack_key': $scope.account.browserstack_key,
+      };
+
+      Account.setBrowserStackData(data);
+      console.log($scope.account.browserstack_username);
```
Comment (amitaibu) [inline client/app/scripts/controllers/account.js]: remove

File: shoov/modules/custom/shoov_restful/plugins/restful/user/config/1.0/ShoovConfigResource.class.php [hunk 2]
```diff
@@ -26,6 +27,10 @@ public function publicFieldsInfo() {
       'callback' => array($this, 'getAccessToken'),
     );
 
+    $public_fields['csrfToken'] = array(
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/user/config/1.0/ShoovConfigResource.class.php]: no need for csrf in headless - we already use the access token

File: client/app/scripts/services/account.js [hunk 3]
```diff
@@ -50,6 +50,48 @@ angular.module('clientApp')
...
+     *
+     * @param data
+     *  The data to set
+     *
+     * @returns {$q.promise}
+     */
+    this.setBrowserStackData = function(data) {
+      var deferred = $q.defer();
+      var url = Config.backend + '/api/config/';
+      this.getAccountConfigData().then(function(val) {
```
Comment (amitaibu) [inline client/app/scripts/services/account.js]: remove

File: client/app/scripts/services/account.js [hunk 4]
```diff
@@ -50,6 +50,48 @@ angular.module('clientApp')
...
+        var csrfToken = val.csrfToken;
+        $http({
```
Comment (amitaibu) [inline client/app/scripts/services/account.js]: `return $http`

File: shoov/modules/custom/shoov_restful/plugins/restful/user/config/1.0/ShoovConfigResource.class.php [hunk 5]
```diff
@@ -13,6 +13,7 @@ class ShoovConfigResource extends \RestfulEntityBaseUser {
   protected $controllers = array(
     '' => array(
       \RestfulInterface::GET => 'viewEntity',
+      \RestfulInterface::PATCH => 'patchEntity',
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/user/config/1.0/ShoovConfigResource.class.php]: use the `me` resource

**Comment by HelenaEksler** [comment]
@amitaibu PR is ready.

File: shoov/modules/custom/shoov_restful/plugins/restful/user/config/1.0/ShoovConfigResource.class.php [hunk 6]
```diff
@@ -26,19 +26,13 @@ public function publicFieldsInfo() {
...
-    $public_fields['browserstack_key'] = array(
-      'property' => 'field_browserstack_key',
-    );
-
     unset($public_fields['id']);
     unset($public_fields['self']);
     unset($public_fields['label']);
     unset($public_fields['url']);
     unset($public_fields['mail']);
+    unset($public_fields['browserstack_username']);
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/user/config/1.0/ShoovConfigResource.class.php]: those fields need to still be kept here - we might have a follow up to merge `Config` and `Account` together

**Comment by amitaibu** [comment]
Merged.

**Pull request closed without merging**